docs/how-to/how-to-use-virtio-fs-nydus-with-kata.md

```diff
@@ -0,0 +1,17 @@
+# Kata Containers with virtio-fs-nydus
+
+## Introduction
+
+Research shows that time to take for pull operation accounts for 76% of container startup time but only 6.4% of that data is read. so if we can get data on demand(lazyload), it will speed up the container start. [Nydusd](https://github.com/dragonflyoss/image-service) is the project which build image with new format and can get data on demand when container start.
```

Comment:
AFAICT, the quoted research applies to traditional containers runtimes. Container startup time is significantly worse in Kata due to VM boot time (although there are methods to mitigate that), so I am pretty convinced that the balance would be quite different for Kata containers. So I would not quote the 76% without some additional qualification.

The idea is appealing, but I wonder if this is not something that would be even more relevant guest-side in the context of the new secure image download we are putting in place for confidential containers (CC), because in that case, there is much less opportunity for reusing already downloaded images, so a lazy loading optimization might pay off more.

The problem, however, is that in the CC case, it might be much more difficult to fetch container contents piece by piece, since now we are talking about encrypted or checksummed images, something which may be difficult to do piece by piece.

Reply:
@c3d CC is surely under our radar. We are working on a design for the CC use case to enable lazy loading and rootfs sharing with nydus. Right now we only have some rough ideas. Will share more details once there is a relatively solid design.

OTOH, the PR is to enable native nydus support for Kata in general. CC will be the next step.